### ISSUE TITLE
[BUG] - ReadWriteMany (RWX) workloads not present in the Longhorn web interface

### ISSUE THREAD

**Issue by Raboldar** [longhorn/longhorn#2009]
After installing the NFS provisioner according to the documentation : 
**01-security.yaml**
```
apiVersion: policy/v1beta1
kind: PodSecurityPolicy
metadata:
  name: longhorn-nfs-provisioner
spec:
  fsGroup:
    rule: RunAsAny
  allowedCapabilities:
    - DAC_READ_SEARCH
    - SYS_RESOURCE
  runAsUser:
    rule: RunAsAny
  seLinux:
    rule: RunAsAny
  supplementalGroups:
    rule: RunAsAny
  volumes:
    - configMap
    - downwardAPI
    - emptyDir
    - persistentVolumeClaim
    - secret
    - hostPath
---
kind: ClusterRole
apiVersion: rbac.authorization.k8s.io/v1
metadata:
  name: longhorn-nfs-provisioner
rules:
  - apiGroups: [""]
    resources: ["persistentvolumes"]
    verbs: ["get", "list", "watch", "create", "delete"]
  - apiGroups: [""]
    resources: ["persistentvolumeclaims"]
    verbs: ["get", "list", "watch", "update"]
  - apiGroups: ["storage.k8s.io"]
    resources: ["storageclasses"]
    verbs: ["get", "list", "watch"]
  - apiGroups: [""]
    resources: ["events"]
    verbs: ["create", "update", "patch"]
  - apiGroups: [""]
    resources: ["services", "endpoints"]
    verbs: ["get"]
  - apiGroups: ["extensions"]
    resources: ["podsecuritypolicies"]
    resourceNames: ["nfs-provisioner"]
    verbs: ["use"]
---
kind: ClusterRoleBinding
apiVersion: rbac.authorization.k8s.io/v1
metadata:
  name: longhorn-nfs-provisioner
subjects:
  - kind: ServiceAccount
    name: longhorn-nfs-provisioner
    namespace: longhorn-system
roleRef:
  kind: ClusterRole
  name: longhorn-nfs-provisioner
  apiGroup: rbac.authorization.k8s.io
---
kind: Role
apiVersion: rbac.authorization.k8s.io/v1
metadata:
  name: leader-locking-longhorn-nfs-provisioner
  namespace: longhorn-system
rules:
  - apiGroups: [""]
    resources: ["endpoints"]
    verbs: ["get", "list", "watch", "create", "update", "patch"]
---
kind: RoleBinding
apiVersion: rbac.authorization.k8s.io/v1
metadata:
  name: leader-locking-longhorn-nfs-provisioner
  namespace: longhorn-system
subjects:
  - kind: ServiceAccount
    name: longhorn-nfs-provisioner
    namespace: longhorn-system
roleRef:
  kind: Role
  name: leader-locking-longhorn-nfs-provisioner
  apiGroup: rbac.authorization.k8s.io

```
***02-longhorn-nfs-provisioner.yaml***
```
apiVersion: v1
kind: ServiceAccount
metadata:
  name: longhorn-nfs-provisioner
  namespace: longhorn-system
---
kind: Service
apiVersion: v1
metadata:
  name: longhorn-nfs-provisioner
  namespace: longhorn-system
  labels:
    app: longhorn-nfs-provisioner
spec:
  # hardcode a cluster ip for the service
  # so that on delete & recreate of the service the previous pv's still point
  # to this nfs-provisioner, pick a new ip for each new nfs provisioner
  clusterIP: 192.168.1.1 
  ports:
    - name: nfs
      port: 2049
    - name: nfs-udp
      port: 2049
      protocol: UDP
    - name: nlockmgr
      port: 32803
    - name: nlockmgr-udp
      port: 32803
      protocol: UDP
    - name: mountd
      port: 20048
    - name: mountd-udp
      port: 20048
      protocol: UDP
    - name: rquotad
      port: 875
    - name: rquotad-udp
      port: 875
      protocol: UDP
    - name: rpcbind
      port: 111
    - name: rpcbind-udp
      port: 111
      protocol: UDP
    - name: statd
      port: 662
    - name: statd-udp
      port: 662
      protocol: UDP
  selector:
    app: longhorn-nfs-provisioner
---
kind: Deployment
apiVersion: apps/v1
metadata:
  name: longhorn-nfs-provisioner
  namespace: longhorn-system
spec:
  selector:
    matchLabels:
      app: longhorn-nfs-provisioner
  replicas: 1
  strategy:
    type: Recreate
  template:
    metadata:
      labels:
        app: longhorn-nfs-provisioner
    spec:
      serviceAccount: longhorn-nfs-provisioner
      containers:
        - name: longhorn-nfs-provisioner
          image: quay.io/kubernetes_incubator/nfs-provisioner:latest
          ports:
            - name: nfs
              containerPort: 2049
            - name: nfs-udp
              containerPort: 2049
              protocol: UDP
            - name: nlockmgr
              containerPort: 32803
            - name: nlockmgr-udp
              containerPort: 32803
              protocol: UDP
            - name: mountd
              containerPort: 20048
            - name: mountd-udp
              containerPort: 20048
              protocol: UDP
            - name: rquotad
              containerPort: 875
            - name: rquotad-udp
              containerPort: 875
              protocol: UDP
            - name: rpcbind
              containerPort: 111
            - name: rpcbind-udp
              containerPort: 111
              protocol: UDP
            - name: statd
              containerPort: 662
            - name: statd-udp
              containerPort: 662
              protocol: UDP
          securityContext:
            capabilities:
              add:
                - DAC_READ_SEARCH
                - SYS_RESOURCE
          args:
            - "-provisioner=nfs.longhorn.io"
            - "-device-based-fsids=false"
          env:
            - name: POD_IP
              valueFrom:
                fieldRef:
                  fieldPath: status.podIP
            - name: SERVICE_NAME
              value: longhorn-nfs-provisioner
            - name: POD_NAMESPACE
              valueFrom:
                fieldRef:
                  fieldPath: metadata.namespace
          imagePullPolicy: "IfNotPresent"
          readinessProbe:
            exec:
              command:
                - ls
                - /export
            initialDelaySeconds: 5
            periodSeconds: 5
          livenessProbe:
            exec:
              command:
                - ls
                - /export
            initialDelaySeconds: 5
            periodSeconds: 5
          volumeMounts:
            - name: export-volume
              mountPath: /export
      volumes:
        - name: export-volume
          persistentVolumeClaim:
            claimName: longhorn-nfs-provisioner
      # we want really quick failover
      terminationGracePeriodSeconds: 30
      tolerations:
        - effect: NoExecute
          key: node.kubernetes.io/not-ready
          operator: Exists
          tolerationSeconds: 60
        - effect: NoExecute
          key: node.kubernetes.io/unreachable
          operator: Exists
          tolerationSeconds: 60
---
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  name: longhorn-nfs-provisioner # longhorn backing pvc
  namespace: longhorn-system
spec:
  storageClassName: longhorn
  accessModes:
    - ReadWriteOnce
  resources:
    requests:
      storage: "35G" # make this 10% bigger then the workload pvc
---
apiVersion: storage.k8s.io/v1
kind: StorageClass
metadata:
  name: longhorn-nfs # workload storage class
provisioner: nfs.longhorn.io
mountOptions:
  - "vers=4.1"
  - "noresvport"
```
*** 03-rwx-test.yaml ***
```
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  name: nfs-test
spec:
  accessModes:
    - ReadWriteMany
  storageClassName: longhorn-nfs
  resources:
    requests:
      storage: 1Gi
---
apiVersion: apps/v1
kind: Deployment
metadata:
  name: nfs-test
  labels:
    app: nfs-test
spec:
  replicas: 4
  selector:
    matchLabels:
      app: nfs-test
  strategy:
    type: Recreate
  template:
    metadata:
      labels:
        app: nfs-test
    spec:
      containers:
        - image: ubuntu:xenial
          imagePullPolicy: Always
          command: ["/bin/sh", "-c"]
          args:
            - sleep 30; touch /mnt/nfs-test/test.log; while true; do echo $hostname >> /mnt/nfs-test/test.log; sleep 1; done;
          name: nfs-test
          stdin: true
          tty: true
          livenessProbe:
            exec:
              command:
                - timeout
                - "10"
                - ls
                - /mnt/nfs-test
            initialDelaySeconds: 10
            periodSeconds: 10
            timeoutSeconds: 10
          volumeMounts:
            - mountPath: /mnt/nfs-test
              name: nfs-test
      restartPolicy: Always
      volumes:
        - name: nfs-test
          persistentVolumeClaim:
            claimName: nfs-test
```
The PVC/PV that was created does not apear in the longhorn interface.
Is there a way to add them ? (label, annotation etc ?)
I suppose the problem comes from the fact that the interface might check only the volumes that have 

**To Reproduce**
Deploy the files in a Longhorn cluster and check the created volumes in the Longhorn web interface. 

**Expected behavior**
The volumes won't be present.

**Log**
Started the kubetail before creating the volume : 
```
[longhorn-manager-z79qp] time="" level=debug msg="websocket: close 1001 (going away)" id=8934703334344090102 type=volumes 
[longhorn-manager-z79qp] time="" level=debug msg="websocket: close 1001 (going away)" id=5770238316347331227 type=engineimages 
[longhorn-manager-z79qp] 10.30.5. - - [] "GET /v1/ws/1s/engineimages HTTP/1.1" 200 0 "" "Mozilla/5.0 (X11; Linux x86_64; rv:78.0) Gecko/20100101 Firefox/78.0" 
[longhorn-manager-z79qp] 10.30.5. - - [] "GET /v1/ws/1s/nodes HTTP/1.1" 200 0 "" "Mozilla/5.0 (X11; Linux x86_64; rv:78.0) Gecko/20100101 Firefox/78.0" 
[longhorn-manager-z79qp] 10.30.5. - - [] "GET /v1/ws/1s/events HTTP/1.1" 200 0 "" "Mozilla/5.0 (X11; Linux x86_64; rv:78.0) Gecko/20100101 Firefox/78.0" 
[longhorn-manager-z79qp] 10.30.5. - - [] "GET /v1/ws/1s/settings HTTP/1.1" 200 0 "" "Mozilla/5.0 (X11; Linux x86_64; rv:78.0) Gecko/20100101 Firefox/78.0" 
[longhorn-manager-z79qp] time="" level=debug msg="websocket: close 1001 (going away)" id=9098028164448970470 type=nodes 
[longhorn-manager-z79qp] time="" level=debug msg="websocket: close 1001 (going away)" id=4066514490795870641 type=events 
[longhorn-manager-z79qp] time="" level=debug msg="websocket: close 1001 (going away)" id=7425666429733388070 type=settings 
[longhorn-manager-z79qp] time="" level=debug msg="websocket: open" id=2725969259462161746 type=nodes 
[longhorn-manager-z79qp] time="" level=debug msg="websocket: open" id=4529645464307011968 type=settings 
[longhorn-manager-z79qp] time="" level=debug msg="websocket: open" id=8649457385099476674 type=events 
[longhorn-manager-z79qp] time="" level=debug msg="websocket: open" id=7144872486935865861 type=engineimages 
[longhorn-manager-z79qp] time="" level=debug msg="websocket: open" id=5292174187434625617 type=volumes 
[longhorn-manager-z79qp] 10.30.5. - - [23/Nov/2020:22:35:55 +0000] "GET /v1/ws/1s/volumes HTTP/1.1" 200 0 "" "Mozilla/5.0 (X11; Linux x86_64; rv:78.0) Gecko/20100101 Firefox/78.0" 
[longhorn-manager-z79qp] 10.30.5. - - [23/Nov/2020:22:52:32 +0000] "GET /v1/ws/1s/settings HTTP/1.1" 200 0 "" "Mozilla/5.0 (X11; Linux x86_64; rv:78.0) Gecko/20100101 Firefox/78.0" 
[longhorn-manager-z79qp] 10.30.5. - - [23/Nov/2020:22:52:32 +0000] "GET /v1/ws/1s/nodes HTTP/1.1" 200 0 "" "Mozilla/5.0 (X11; Linux x86_64; rv:78.0) Gecko/20100101 Firefox/78.0" 
[longhorn-manager-z79qp] time="" level=debug msg="websocket: close 1001 (going away)" id=4529645464307011968 type=settings 
[longhorn-manager-z79qp] time="" level=debug msg="websocket: close 1001 (going away)" id=2725969259462161746 type=nodes 
[longhorn-manager-z79qp] time="" level=debug msg="websocket: close 1001 (going away)" id=8649457385099476674 type=events 
[longhorn-manager-z79qp] time="" level=debug msg="websocket: close 1001 (going away)" id=7144872486935865861 type=engineimages 
[longhorn-manager-z79qp] 10.30.5. - - [23/Nov/2020:22:52:32 +0000] "GET /v1/ws/1s/events HTTP/1.1" 200 0 "" "Mozilla/5.0 (X11; Linux x86_64; rv:78.0) Gecko/20100101 Firefox/78.0" 
[longhorn-manager-z79qp] 10.30.5. - - [23/Nov/2020:22:52:32 +0000] "GET /v1/ws/1s/engineimages HTTP/1.1" 200 0 "" "Mozilla/5.0 (X11; Linux x86_64; rv:78.0) Gecko/20100101 Firefox/78.0" 
[longhorn-manager-z79qp] 10.30.5. - - [23/Nov/2020:22:52:32 +0000] "GET /v1/ws/1s/volumes HTTP/1.1" 200 0 "" "Mozilla/5.0 (X11; Linux x86_64; rv:78.0) Gecko/20100101 Firefox/78.0" 
[longhorn-manager-z79qp] time="" level=debug msg="websocket: close 1001 (going away)" id=5292174187434625617 type=volumes 
[longhorn-manager-z79qp] time="" level=debug msg="websocket: open" id=1290673718745737673 type=nodes 
[longhorn-manager-z79qp] time="" level=debug msg="websocket: open" id=5699327448539958727 type=settings 
[longhorn-manager-z79qp] time="" level=debug msg="websocket: open" id=5071102846248741684 type=events 
[longhorn-manager-z79qp] time="" level=debug msg="websocket: open" id=288953509856197651 type=engineimages 
[longhorn-manager-z79qp] time="" level=debug msg="websocket: open" id=4111475331035616120 type=volumes 

```

**Environment:**
 - Longhorn version: 1.0.2
 - Kubernetes version: v1.19.3
 - Node config
   - OS type and version: Debian 10 Buster
   - Disk type: HDD
 - Underlying Infrastructure (e.g. on AWS/GCE, EKS/GKE, VMWare/KVM, Baremetal): Baremetal, Longhorn installed using Rancher.

The volume is created ad it works, the only problem is that it isn't added to the interface.

**Comment by khushboo-rancher** [comment]
@Raboldar If I understood your problem statement correctly, you are not seeing the volume pertaining to PVC `nfs-test` on the longhorn UI?

In the above yaml, volume pertaining to PVC `nfs-test` is **not** expected to be present in the Longhorn UI as explained below.

With the above yaml, a volume attached to `longhorn-nfs-provisioner` is created and should be available on the Longhorn UI. The pods of the deployment `nfs-test` access the same volume (RWX) which is available inside the volume attached  to `longhorn-nfs-provisioner`  as a directory with the PV name. You can check the mount point of the volume attached to `longhorn-nfs-provisioner` which is `/export` and see the directory getting created with the PV name.

**Comment by Raboldar** [comment]
Thanks for the reply !!
I understand why the pods don't appear in the interface but is there any way to add them ? (label, annotation..)
It would be great to be able to monitor all the disks with the same interface (Longhorn web). At worst I could do a script that would print infos about the nsf disks but it would be great so somehow integrate them in the interface.
If it isn't possible for now could this be a feature proposal ?

**Comment by khushboo-rancher** [comment]
We have a feature planned for this already https://github.com/longhorn/longhorn/issues/1849 and this will be available in Longhorn v1.1.0 coming soon.

**Comment by Raboldar** [comment]
Glad to hear ! Thanks for the help !

**Comment by khushboo-rancher** [comment]
@Raboldar This feature has been released as part of Longhorn v1.1.0. You can refer https://longhorn.io/docs/1.1.0/advanced-resources/rwx-workloads/ to try it out and you should be able to view RWX volumes on Longhorn UI.

**Comment by joshimoo** [comment]
Closing this since we released native rwx support in v1.1 
https://longhorn.io/docs/1.1.0/advanced-resources/rwx-workloads/
